rspirv/lift/mod.rs

```diff
@@ -93,23 +132,60 @@ impl LiftContext {
             )?;
             //TODO: lift function type instruction
 
-            let mut basic_blocks = Vec::with_capacity(fun.basic_blocks.len());
-            for block in fun.basic_blocks.iter() {
-                basic_blocks.push(module::BasicBlock {
-                    ops: Vec::new(),
-                    terminator: context.lift_terminator(
-                        block.instructions
-                            .last()
-                            .ok_or(ConversionError::MissingTerminator)?
-                    )?,
-                });
+            let mut blocks = Vec::with_capacity(fun.blocks.len());
+            for block in fun.blocks.iter() {
+                let mut arguments = Vec::new();
+                for inst in &block.instructions {
+                    match inst.class.opcode {
+                        spirv::Op::Line => {} // skip line decorations
+                        spirv::Op::Phi => {
+                            match inst.operands[0] {
+                                dr::Operand::IdRef(id) => {
+                                    let (_, info) = context.ops.lookup(id);
+                                    arguments.push(info.ty
+                                        .expect("No return type for Phi source?")
```

Comment:
A question mark in an error message is a bit odd, also is this panicking if there's an error in the SPIR-V? That seems like a dangerous road to go down.

Reply:
good point, I'm fixing this